src/utils/utils.ts

```diff
@@ -90,24 +90,44 @@ function createApp(
 
   const gitBranch = repoUrl.includes('template_sasonly') ? 'master' : 'main'
 
+  const gitVersion: string = (shelljs
+    .exec(`git version`, { silent: true })
+    .stdout.match(/(?<!git version)\d+\.\d+\.\d+/) || '')[0].replace(/\./g, '')
+
   shelljs.exec(
-    `cd "${folderPath}" && git clone --depth 1 -b ${gitBranch} ${repoUrl} .`,
+    `cd "${folderPath}" && git clone --recurse${
+      parseInt(gitVersion) > 2130 ? '-submodules' : ''
```

Comment:
A comment to explain why this section is necessary would be good

Reply:
good point

Reply:
done